api/redhatopenshift/HcpCluster/hcpCluster-models.tsp

```diff
@@ -218,6 +214,21 @@ model ApiProfile {
   visibility: Visibility;
 }
 
+/** Information about how Etcd is Encrypted. */
+model EtcdEncryptionProfile {
```

Comment:
As a side note, it seems that for ROSA-HCP we *always* enforce etcd encryption at data level. In case the keys are not provided it seems we are generating some.

I've been exploring why is that the case and I found the following Jira issue: https://issues.redhat.com/browse/OCM-3842 that we should investigate to understand it and figure out what do we need to do for ARO-HCP around this. The following is mentioned on it:

> Our storage of etcd snapshots requires that we ensure the etcd records are always encrypted. We enable customers to supply their own KMS key for etcd encryption, but in the absence of that key, we must fall back to ensuring we still rely on AES encryption of etcd. We cannot enable a flag to allow customers to disable this without also enabling them to disable etcd backups